File: Python/checkoutpage_find_by_merchant_reference_number.py
# Description: Checkoutpage Find by Merchant Reference Number request sample for Checkout Page API.
import requests
import json

url = "https://checkout.bmstaging.id/api/findByRef/1257582293"

payload={}
headers = {
  'X-Winpay-Timestamp': '2026-11-24T17:07:05+07:00',
  'X-Winpay-Key': 'xxxxxxxx',
  'X-Winpay-Signature': 'xxxxxxxxxxxxxxxxx',
  'Content-Type': 'application/json',
  'Accept': 'application/json'
}

response = requests.request("GET", url, headers=headers, data=payload)

print(response.text)

File: Ruby/checkoutpage_find_by_merchant_reference_number.rb
# Description: Checkoutpage Find by Merchant Reference Number request sample for Checkout Page API.
require "uri"
require "json"
require "net/http"

url = URI("https://checkout.bmstaging.id/api/findByRef/1257582293")

https = Net::HTTP.new(url.host, url.port)
https.use_ssl = true

request = Net::HTTP::Get.new(url)
request["X-Winpay-Timestamp"] = "2026-11-24T17:07:05+07:00"
request["X-Winpay-Key"] = "xxxxxxxx"
request["X-Winpay-Signature"] = "xxxxxxxxxxxxxxxxx"
request["Content-Type"] = "application/json"
request["Accept"] = "application/json"

response = https.request(request)
puts response.read_body
